### PR TITLE
feat: adding youtube & vimeo

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/6479
 - [ ] server PR

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Please add screenshots

File: app/styles/components/extras.scss [hunk 1]
```diff
@@ -28,3 +28,9 @@ a.link-item {
     font-size: .9rem;
   }
 }
+
+.stream {
+  height: 80%;
+  margin-left: 15% !important;
+  width: 70%;
+}
```
Comment (iamareebjamal) [inline app/styles/components/extras.scss:36]: Why this?

File: app/components/public/stream/video-stream.hbs [hunk 2]
```diff
@@ -8,5 +8,15 @@
   {{#if (eq @videoStream.videoChannel.provider 'jitsi')}}
     <Public::Stream::JitsiStream @videoStream={{@videoStream}} />
   {{/if}}
+  {{#if (eq @videoStream.videoChannel.provider 'youtube')}}
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:11]: Shouldn't you check for this.youtubeId?

File: app/components/public/stream/video-stream.hbs [hunk 3]
```diff
@@ -8,5 +8,15 @@
...
+    <div class="ui segment mt-4 {{if device.isMobile '' 'stream'}}">
+      <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="http://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
+    </div>
+  {{/if}}
+  {{#if (eq @videoStream.videoChannel.provider 'vimeo')}}
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:16]: Same here for vimeoId

**Comment by iamareebjamal** [comment]
Screenshots?

File: app/components/public/stream/video-stream.hbs [hunk 4]
```diff
@@ -8,5 +8,15 @@
   {{#if (eq @videoStream.videoChannel.provider 'jitsi')}}
     <Public::Stream::JitsiStream @videoStream={{@videoStream}} />
   {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'youtube') this.youtubeId)}}
+    <div class="ui segment mt-4 {{if device.isMobile '' 'stream'}}">
+      <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="http://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:13]: Why loading http iframe? Instead of https?

Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:13]:
```suggestion
      <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
```

**Comment by iamareebjamal** [comment]
The video should autoplay and there is a weird padding around the video. It should be stretched across screen

File: app/components/public/stream/video-stream.hbs [hunk 5]
```diff
@@ -8,5 +8,15 @@
   {{#if (eq @videoStream.videoChannel.provider 'jitsi')}}
     <Public::Stream::JitsiStream @videoStream={{@videoStream}} />
   {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'youtube') this.youtubeId)}}
+    <div class="ui segment mt-4 {{if device.isMobile '' 'stream'}}">
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:12]: Why are you adding segment and mt-4? It doesn't make any sense?

See how BBB iframe is added!

File: app/components/public/stream/video-stream.hbs [hunk 6]
```diff
@@ -8,5 +8,15 @@
...
+      <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
+    </div>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:14]:
```suggestion
    <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
```

File: app/components/public/stream/video-stream.hbs [hunk 7]
```diff
@@ -8,5 +8,15 @@
...
+  {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'vimeo') this.vimeoId)}}
+    <div class="ui segment mt-4 {{if device.isMobile '' 'stream'}}">
+      <iframe src="https://player.vimeo.com/video/{{this.vimeoId}}" width="100%" height="100%" frameborder="0" title="Vimeo Live Stream" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
+    </div>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:19]:
```suggestion
    <iframe src="https://player.vimeo.com/video/{{this.vimeoId}}" width="100%" height="100%" frameborder="0" title="Vimeo Live Stream" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
```

File: app/components/public/stream/video-stream.hbs [hunk 8]
```diff
@@ -8,5 +8,11 @@
   {{#if (eq @videoStream.videoChannel.provider 'jitsi')}}
     <Public::Stream::JitsiStream @videoStream={{@videoStream}} />
   {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'youtube') this.youtubeId)}}
+    <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:12]:
```suggestion
    <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1&?autoplay=1" gesture="media" allow="encrypted-media" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
```

File: app/components/public/stream/video-stream.hbs [hunk 9]
```diff
@@ -8,5 +8,11 @@
...
+  {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'vimeo') this.vimeoId)}}
+    <iframe src="https://player.vimeo.com/video/{{this.vimeoId}}" width="100%" height="100%" frameborder="0" title="Vimeo Live Stream" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:15]:
```suggestion
    <iframe src="https://player.vimeo.com/video/{{this.vimeoId}}" width="100%" height="100%" frameborder="0" title="Vimeo Live Stream" allowfullscreen="true"></iframe>
```

**Comment by maze-runnar** [comment]
Screenshot from 2021-02-11 10-12-38
Screenshot from 2021-02-11 10-12-34
Screenshot from 2021-02-11 10-12-10
Screenshot from 2021-02-11 10-12-06

**Comment by iamareebjamal** [comment]
See the above comments. The padding doesn't make sense. You should compare existing integrations like BBB and Jitsi to realize that there should be no padding or margins at all. You could see how BBB iframe was added and just copied it. 

I have fixed it now

File: app/components/public/stream/video-stream.hbs [hunk 10]
```diff
@@ -8,5 +8,11 @@
   {{#if (eq @videoStream.videoChannel.provider 'jitsi')}}
     <Public::Stream::JitsiStream @videoStream={{@videoStream}} />
   {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider 'youtube') this.youtubeId)}}
+    <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1&?autoplay=1" gesture="media" allow="encrypted-media" allowfullscreen="true"></iframe>
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:12]:
```suggestion
    <iframe width="100%" height="100%" frameborder="0" title="YouTube Live Stream" src="https://www.youtube.com/embed/{{this.youtubeId}}?enablejsapi=1&autoplay=1" gesture="media" allow="encrypted-media" allowfullscreen="true"></iframe>
```

File: app/styles/components/extras.scss [hunk 11]
```diff
@@ -28,3 +28,8 @@ a.link-item {
     font-size: .9rem;
   }
 }
+
+.stream {
+  height: 100%;
+  width: 100%;
+}
```
Comment (iamareebjamal) [inline app/styles/components/extras.scss:35]:
```suggestion

```

File: app/components/public/stream/video-stream.ts [hunk 12]
```diff
@@ -35,6 +41,19 @@ export default class PublicStreamVideoStream extends Component<Args> {
 
     if (provider === 'jitsi') {
       this.loading = false;
+    } else if (provider === 'youtube') {
+      const match = stream.url.split('v=');
+      if (match[1]) {
+        this.youtubeId = match[1];
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.ts:47]:
```suggestion
      if (id) {
        this.youtubeId = id;
```

File: app/components/public/stream/video-stream.ts [hunk 13]
```diff
@@ -35,6 +41,19 @@ export default class PublicStreamVideoStream extends Component<Args> {
...
+      }
+      this.loading = false;
+    } else if (provider === 'vimeo') {
+      const regExp = /https:\/\/(www\.)?vimeo.com\/(\d+)($|\/)/;
+      const match = stream.url.match(regExp);
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.ts:52]:
```suggestion
      const [,,id] = stream.url.match(regExp);
```

**Pull request merged**